### DISCUSSION TITLE
A 'ReporterPointSource' is not a registered object

### DISCUSSION THREAD

**Discussion by aaelmeli** [idaholab/moose#18845] [Q&A General] [answered]
Hi, I am trying to apply point source by using Dirac kernel that read the point sources from csv file. I am using the ReporterPointSource object to do that but I am getting an error. This error is as follows:

> *** ERROR ***
A 'ReporterPointSource' is not a registered object.

>If you are trying to find this object in a dynamically linked library, make sure that
the library can be found either in your "Problem/library_path" parameter or in the
MOOSE_LIBRARY_PATH environment variable.

>Stack frames: 21
0: libMesh::print_trace(std::ostream&)
1: moose::internal::mooseErrorRaw(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)
2: void mooseError<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&&)
3: Factory::reportUnregisteredError(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const
4: Factory::getValidParams(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
5: MooseObjectAction::MooseObjectAction(InputParameters)
6: AddDiracKernelAction::AddDiracKernelAction(InputParameters)
7: /home/aaelmeli/projects/moose/framework/libmoose-opt.so.0(+0x7c9b98) [0x7fa6369f9b98]
8: std::shared_ptr<Action> moose::internal::buildAct<AddDiracKernelAction>(InputParameters const&)
9: ActionFactory::create(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, InputParameters&)
10: Parser::walkRaw(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, hit::Node*)
11: Parser::walk(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, hit::Node*)
12: hit::Node::walk(hit::Walker*, hit::NodeType)
13: hit::Node::walk(hit::Walker*, hit::NodeType)
14: hit::Node::walk(hit::Walker*, hit::NodeType)
15: Parser::parse(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&)
16: MooseApp::setupOptions()
17: MooseApp::run()
18: ./first_test-opt(+0x5263) [0x7fa63800d263]
19: __libc_start_main
20: ./first_test-opt(+0x54ed) [0x7fa63800d4ed]
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
[unset]: write_line error; fd=-1 buf=:cmd=abort exitcode=1
:
system msg for write_line failure : Bad file descriptor

My minimal input file reproduce this error is as follows:

 ```
[Mesh]
   type = GeneratedMesh
    dim = 2
     xmin=0
    xmax=1
    ymin=0
    ymax=1.0
    nx = 10
    ny=10
[]

[Variables]
  [u]
    order = FIRST
    family = LAGRANGE
  []
[]


[Kernels]
  [diff]
    type = MatDiffusion
    variable = u
    diffusivity = _diff2_
  []
[]

[DiracKernels]

  [./diracforce]
  type = ReporterPointSource
   variable = u
   value_name = 'csv_reader/u'
   x_coord_name = csv_reader/x
   y_coord_name = csv_reader/y
    z_coord_name = csv_reader/z
  []
[]


[Functions]
    [./func]
    type = ParsedFunction
    value=10
    [../]
[]
[Materials]
   [./mat_parsed]
   type = GenericFunctionMaterial
   prop_names  = '_diff2_'
   prop_values = 'func'
   [../]
[]

[BCs]
  [./BC_left]
    type = DirichletBC
    variable = u
    boundary = 3
    value= 0
  []
  [./BC_right]
    type = NeumannBC
    variable = u
    boundary = 1
    value= 1000
  []
[]

[VectorPostprocessors]
    [csv_reader]
    type = CSVReader
    csv_file = 'zForwardOutput/all_u_0002.csv'
  []
[]
[Executioner]
execute_on = 'initial timestep_end'
  type = Steady
  solve_type = 'LINEAR'
[]


[Outputs]
  [exodus]
    file_base = 'ForwardOutput/out_'
    type = Exodus
    execute_on = final
  []
  [all_response]
    file_base = 'ForwardOutput/all_'
    type = CSV
    execute_vector_postprocessors_on = initial
  []
[]


```
Any idea on how to fix this?

**Comment by GiudGiud** [comment]
Hello

I cant see what's wrong. This is a moose object so it should be always included. 
Are you using a really old version of MOOSE?

~~Can you try moving this DiracKernel to another input and see if it works there? Just in case there's something going on with the formatting.~~

Guillaume

**Comment by dschwen** [comment, marked as answer]
That object was added about a month ago. Can you make sure your MOOSE is up to date?

**Reply by aaelmeli** [reply]
It seems not up-to-date, yes. However, when I updated it to the latest version. I could not build it. I am getting the compiler errors below (in bold).

> Using HIT from /home/aaelmeli/projects/moose/framework/contrib/hit
Compiling C++ (in opt mode) /home/aaelmeli/projects/moose/framework/build/unity_src/auxkernels_Unity.C...
Compiling C++ (in opt mode) /home/aaelmeli/projects/moose/framework/build/unity_src/controls_Unity.C...
Compiling C++ (in opt mode) /home/aaelmeli/projects/moose/framework/build/unity_src/dampers_Unity.C...
Compiling C++ (in opt mode) /home/aaelmeli/projects/moose/framework/build/unity_src/dgkernels_Unity.C...
In file included from /home/aaelmeli/projects/moose/framework/build/unity_src/auxkernels_Unity.C:50:
/home/aaelmeli/projects/moose/framework/src/auxkernels/SolutionAux.C: In member function 'virtual libMesh::Real SolutionAux::computeValue()':
/home/aaelmeli/projects/moose/framework/src/auxkernels/SolutionAux.C:97:63: **error**: 'const class libMesh::Elem' has no member named 'vertex_average'
   97 |       output = _solution_object.pointValue(_t, _current_elem->**vertex_average()**, _var_name);
      |                                                               ^~~~~~~~~~~~~~
In file included from /home/aaelmeli/projects/moose/framework/build/unity_src/auxkernels_Unity.C:52:
/home/aaelmeli/projects/moose/framework/src/auxkernels/SpatialUserObjectAux.C: In member function 'virtual libMesh::Real SpatialUserObjectAux::computeValue()':
/home/aaelmeli/projects/moose/framework/src/auxkernels/SpatialUserObjectAux.C:41:53: **error**: 'const class libMesh::Elem' has no member named 'vertex_average'
   41 |     return _user_object.spatialValue(_current_elem->**vertex_average()**);
      |                                                     ^~~~~~~~~~~~~~
Compiling C++ (in opt mode) /home/aaelmeli/projects/moose/framework/build/unity_src/dirackernels_Unity.C...
make: *** [/home/aaelmeli/projects/moose/framework/build.mk:139: /home/aaelmeli/projects/moose/framework/build/unity_src/auxkernels_Unity.x86_64-conda-linux-gnu.opt.lo] Error 1
make: *** Waiting for unfinished jobs....

**Reply by loganharbour** [reply]
You now need to update your libmesh.

If you're using conda:
```
conda update --all
```
